src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -564,118 +567,52 @@ uct_ud_mlx5_iface_query(uct_iface_h tl_iface, uct_iface_attr_t *iface_attr)
 }
 
 static ucs_status_t
-uct_ud_mlx5_ep_create_ah(uct_ud_mlx5_iface_t *iface, uct_ud_mlx5_ep_t *ep,
-                         const uct_ib_address_t *ib_addr, unsigned path_index,
-                         const uct_ud_iface_addr_t *if_addr)
-{
+uct_ud_mlx5_iface_get_peer_address(uct_ud_iface_t *iface,
+                                   const uct_ib_address_t *ib_addr,
+                                   const uct_ud_iface_addr_t *if_addr,
+                                   int path_index, void *address_p)
```

Comment:
seems like it should be uct_ud_mlx5_ep_unpack_peer_address(ep, ib_addr, if_addr) ?

Reply:
pls note the arge: instead of iface+pathindex+address_p, can pass ep

Reply:
unfortunately, it couldn't be used in UD/CONN_MATCH adoption - e.g.
https://github.com/dmitrygx/ucx/blob/66eb0c34ca37d95b2ba717dfe58b02b583c2aec8/src/uct/ib/ud/base/ud_iface.c#L54

Reply:
ic, need to use the address for lookup before creating the ep

Reply:
> ic, need to use the address for lookup before creating the ep

yep